roofit/roofitcore/src/RooAbsData.cxx

```diff
@@ -237,10 +238,9 @@ RooAbsData::~RooAbsData()
 
 void RooAbsData::convertToVectorStore()
 {
-  if (dynamic_cast<RooTreeDataStore*>(_dstore)) {
-    RooVectorDataStore* newStore =  new RooVectorDataStore(*(RooTreeDataStore*)_dstore,_vars,GetName()) ;
-    delete _dstore ;
-    _dstore = newStore ;
+   auto tmp = dynamic_cast<RooTreeDataStore *>(&*_dstore);
```

Comment:
There is unique_ptr::get function to do this. But I am still quite puzzled. Why we need this dynamic cast here?

Can't we do `auto tmp = (defaultStorageType==Tree)`?